src/inmanta/data/__init__.py

```diff
@@ -3955,6 +3955,22 @@ def _get_paging_resource_log_item_count_query(
     async def get(cls, action_id: uuid.UUID, connection: Optional[asyncpg.connection.Connection] = None) -> "ResourceAction":
         return await cls.get_one(action_id=action_id, connection=connection)
 
+    async def insert(self, connection: Optional[asyncpg.connection.Connection] = None) -> None:
+        async with self.get_connection(connection) as con:
+            async with con.transaction():
+                await super(ResourceAction, self).insert(con)
+
+                # Also do the join table in the same transaction
+                assert self.resource_version_ids
+                # No additional checking of field validity is done here, because the insert above validates all fields
```

Comment:
it would turn the UUID's into strings, which is weird, because asyncpg does that as well. 

The other two values are produced here as the correct type. 

Should I add the get_value?

Reply:
It does turn the uuid's into strings, but I don't know why

Reply:
I'm in favor unless it has a non-negligible performance impact. I think it's generally a good idea to stick to the pattern we have of always calling it, rather than to make assumptions about the specific circumstance. That said, I don't feel too strongly about this.